platform/ios/src/MGLMapView.mm

```diff
@@ -5997,15 +5997,22 @@ - (CGPoint)userLocationAnnotationViewCenter
 
     // When tracking course, it’s more important to see the road ahead, so
     // weight the user dot down towards the bottom.
-    switch (self.userLocationVerticalAlignment) {
-        case MGLAnnotationVerticalAlignmentCenter:
-            break;
-        case MGLAnnotationVerticalAlignmentTop:
-            center.y = CGRectGetMinY(contentFrame);
-            break;
-        case MGLAnnotationVerticalAlignmentBottom:
-            center.y = CGRectGetMaxY(contentFrame);
-            break;
+//    switch (self.userLocationVerticalAlignment) {
+//        case MGLAnnotationVerticalAlignmentCenter:
+//            break;
+//        case MGLAnnotationVerticalAlignmentTop:
+//            center.y = CGRectGetMinY(contentFrame);
+//            break;
+//        case MGLAnnotationVerticalAlignmentBottom:
+//            center.y = CGRectGetMaxY(contentFrame);
+//            break;
+//    }
+    
+    if ([self.delegate respondsToSelector:@selector(userLocationAnchorForMapView:)])
+    {
+        // Get whatever value is returned from the delegate method that responds to
+        // the selector and assign it to the center
+        center = [self.delegate userLocationAnchorForMapView:self];
```

Comment:
This line sets `center` from scratch, effectively making it relative to the origin rather than the view’s center, which is stored in `center` up to this point.

Reply:
If we choose to set the anchor point based off the the origin, should we adjust the original center (`CGPointMake(CGRectGetMidX(contentFrame), CGRectGetMidY(contentFrame))`) to do the same?

Reply:
No, because the legacy `userLocationVerticalAlignment` code depends on that initial value, whereas the new `-mapViewUserLocationAnchorPoint:` code does not.